Userland/Games/ColorLines/HueFilter.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Math.h>
+#include <LibGfx/Filters/MatrixFilter.h>
+
+class HueFilter : public Gfx::MatrixFilter {
```

Comment:
We already have a `Gfx::HueRotateFilter`, which I think could be used instead of this.

Reply:
Yes, I initially was using the Hue filter provided in the LibGfx library. But, the filter in this game uses a different approach and formula. This is so because the existing LibGfx Hue filter produces the different, darker colors.  The filter in this game produces more lighter colors that better suit the game. The filter formula is taken from https://stackoverflow.com/questions/8507885/shift-hue-of-an-rgb-color and http://www.graficaobscura.com/matrix/index.html . Interesting enough, I was not able to get the "yellow" hue out of both filters, and get yellow marbles, making a total color/hue marble count 7 (the game has 6 different marble colors). But 6 different colors makes more interesting game play in the end.